spring-kafka/src/test/java/org/springframework/kafka/core/DefaultKafkaProducerFactoryTests.java

```diff
@@ -320,12 +454,12 @@ protected Producer createKafkaProducer() {
 		pf.closeThreadBoundProducer();
 		assertThat(KafkaTestUtils.getPropertyValue(pf, "threadBoundProducers", Map.class).get(Thread.currentThread()))
 				.isNull();
-		verify(producer, times(3)).close(any(Duration.class));
+		verify(producer, times(2)).close(any(Duration.class));
```

Comment:
I changed the `assert` because in the refactored `close`/`closeDelegate` methods, there is no need to call close one more time on an already closed producer. I am unsure why it was done previously. If I'm missing something, please let me know.